src/Analysis/Engine/Impl/DependencyResolution/Arc.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Python Tools for Visual Studio
+// Copyright(c) Microsoft Corporation
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the License); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY
+// IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABILITY OR NON-INFRINGEMENT.
+//
+// See the Apache Version 2.0 License for specific language governing
+// permissions and limitations under the License.
+
+using System.Collections.Generic;
+
+namespace Microsoft.PythonTools.Analysis.DependencyResolution {
+    internal partial struct PathResolverSnapshot {
+        /// <summary>
+        /// Represents arc between two nodes in the tree
+        /// </summary>
+        private readonly struct Arc {
+            // List of vertexes.
+            // To provide immutability, it must be changed only by calling List<T>.Add
+            // Better to be replaced with some immutable structure
+            private readonly List<(int index, Node node)> _vertexes;
```

Comment:
Following offline discussion - comment may come handy what is that `index` represents and how it is used since structure is a bit different from typical node with children.

Reply:
Yep, will add it. And add a `DebuggerDisplay` as well.